public/js/controllers/attackcost_controller.js

```diff
@@ -0,0 +1,438 @@
+    }
+  }
+
+  updateTargetHashRate (newTargetPow) {
+    this.targetPowTarget.value = newTargetPow || this.targetPowTarget.value
+
+    switch (this.settings.attack_type) {
+      case 1:
+        this.targetHashRate = hashrate / (1 - parseFloat(this.targetPowTarget.value) / 100) - hashrate
+        this.projectedPriceDivTarget.style.display = 'block'
+        this.internalAttackTextTarget.classList.add('d-none')
+        this.internalAttackPosTextTarget.classList.add('d-none')
+        this.externalAttackTextTarget.classList.remove('d-none')
+        this.externalAttackPosTextTarget.classList.remove('d-node')
+        console.log(this.externalAttackPosTextTarget)
+        return
+      case 0:
+      default:
+        this.targetHashRate = hashrate * parseFloat(this.targetPowTarget.value) / 100
+        this.projectedPriceDivTarget.style.display = 'none'
+        this.externalAttackTextTarget.classList.add('d-none')
+        this.externalAttackPosTextTarget.classList.add('d-node')
+        this.internalAttackTextTarget.classList.remove('d-none')
+        this.internalAttackPosTextTarget.classList.remove('d-none')
+    }
+  }
+
+  updateSliderData () {
+    var val = Math.min(parseFloat(this.attackPercentTarget.value) || 0, 0.99)
+    // Makes PoS to be affected by the slider
+    // Target PoS value increases when slider moves to the right
+    this.setAllInputs(this.targetPosTargets, val * 100)
+
+    this.updateTargetHashRate(val * 100)
+    this.setActivePoint()
+
+    var rate = rateCalculation(val)
+    var randomVal = (rate * this.targetHashRate) / parseFloat(hashrate) * 100
+
+    this.targetPowTarget.value = Math.floor(randomVal * 100) / 100
+    this.internalHashTarget.innerHTML = digitformat((rate * this.targetHashRate), 4) + ' Ph/s '
+    this.ticketsTarget.innerHTML = digitformat(val * tpSize) + ' tickets '
+    switch (this.settings.attack_type) {
+      case 1:
+        this.hideAll(this.internalAttackPosTextTargets)
+        this.showAll(this.externalAttackPosTextTargets)
+        break
+      case 0:
+      default:
+        this.hideAll(this.externalAttackPosTextTargets)
+        this.showAll(this.internalAttackPosTextTargets)
+    }
+
+    this.calculate(true)
+  }
+
+  calculate (disableHashRateUpdate) {
+    if (!disableHashRateUpdate) this.updateTargetHashRate()
+
+    this.targetHashRate *= rateCalculation(this.targetPosTarget.value / 100)
+    this.query.replace(this.settings)
+    var deviceInfo = deviceList[this.selectedDevice()]
+    var deviceCount = Math.ceil((this.targetHashRate * 1000) / deviceInfo.hashrate)
+    var totalDeviceCost = deviceCount * deviceInfo.cost
+    var totalKwh = deviceCount * deviceInfo.power * parseFloat(this.attackPeriodTarget.value) / 1000
+    var totalElectricity = totalKwh * parseFloat(this.kwhRateTarget.value)
+    var extraCost = parseFloat(this.otherCostsTarget.value) / 100 * (totalDeviceCost + totalElectricity)
+    var totalPow = extraCost * totalDeviceCost + totalElectricity
```

Comment:
This seems to be the error. Shouldn't this be `extraCost + totalDeviceCost + totalElectricity`?

If you aren't seeing this error, you should check to make sure your web assets are recompiling.

Reply:
🤦🏻‍♂️
Too bad, I missed this. 
Thank you